src/server/pachyderm_test.go

```diff
@@ -647,6 +647,56 @@ func TestPipelineFailure(t *testing.T) {
 	require.True(t, strings.Contains(jobInfo.Reason, "datum"))
 }
 
+func TestPipelineErrorHandling(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	c := getPachClient(t)
+	require.NoError(t, c.DeleteAll())
+
+	dataRepo := tu.UniqueString("TestPipelineErrorHandling_data")
+	require.NoError(t, c.CreateRepo(dataRepo))
+
+	_, err := c.PutFile(dataRepo, "master", "file1", strings.NewReader("foo\n"))
+	require.NoError(t, err)
+
+	_, err = c.PutFile(dataRepo, "master", "file2", strings.NewReader("bar\n"))
+	require.NoError(t, err)
+
+	pipeline := tu.UniqueString("pipeline")
+	_, err = c.PpsAPIClient.CreatePipeline(
+		context.Background(),
+		&pps.CreatePipelineRequest{
+			Pipeline: client.NewPipeline(pipeline),
+			Transform: &pps.Transform{
+				Cmd:    []string{"bash"},
+				Stdin:  []string{"if", fmt.Sprintf("[ -a pfs/%v/file1 ]", dataRepo), "then", "exit 1", "fi", "ls pfs"},
+				ErrCmd: []string{"bash"},
```

Comment:
I'm a little confused about what this test is doing. Does `Stdin` get applied to both `Cmd` and `ErrCmd`? Reading the code that doesn't seem to be what's happening, but I can't see how else this test would pass, since I think `bash` will just hang if it doesn't get any input, and CI is indeed passing here...

Reply:
I think maybe what's going on is that the datum is actually successfully getting processed and the error handler isn't getting called at all. The check below on line 696 would seem to confirm this.

Also in terms of design I think it's probably pretty weird to have the same `Stdin` getting passed to different commands, so it'd probably make more sense to have an `ErrStdin` or something.

Reply:
It's not, I was planning to ask about making ErrStdin. The intent here was just to choose an ErrCmd which wouldn't do anything besides triggering the recovered datum flow.

Reply:
The pipeline cmd should pass on one datum and fail on the other. Since there is an errcmd, the failed datum should go through the error handling code (which does nothing), and the job should still succeed with just the one datum. So this test is working as I expected.

Reply:
Ahh, this makes sense, turns out I was more than a little confused :grimacing:. I didn't realize that `bash` just returns with a `0` if you give it a `nil` Stdin. Makes sense why this is passing. As a minor tweak how would you feel about using `"true"` as the command? I feel like I'm not the only one who might be confused about this, and that's also probably a good canonical way to have an error handler that ignores all errors.